src/js/stores/AppFormTransforms.js

```diff
@@ -1,9 +1,32 @@
+var Util = require("../helpers/Util");
+
+function trimDuplicableRows(rows) {
+  var trimmedRows = rows.map((row) => {
+    return Object.keys(row).reduce((memo, key) => {
+      if (row[key] != null
+        && key !== "consecutiveKey"
+        && !Util.isEmptyString(row[key])) {
+        memo[key] = row[key];
+      }
+      return memo;
+    }, {});
+  }).filter((row) => Object.keys(row).length);
+
+  if (trimmedRows.length) {
+    return trimmedRows;
+  }
+  return null;
+}
+
 const AppFormTransforms = {
   cpus: (value) => parseFloat(value),
   disk: (value) => parseFloat(value),
   constraints: (constraints) => constraints
     .split(",")
     .map((constraint) => constraint.split(":").map((value) => value.trim())),
+  containerVolumes: (rows) => trimDuplicableRows(rows),
```

Comment:
Instead of excluding keys (```consecutiveKey```), the needed keys should be explicitly included to ensure the object validity. This makes the ```trimDuplicateRows``` kind of obsolete.